File: src/main/java/net/reflact/common/attribute/AttributeRegistry.java
package net.reflact.common.attribute;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class AttributeRegistry {
    private static final Map<String, Attribute> ATTRIBUTES = new HashMap<>();

    public static void register(Attribute attribute) {
        ATTRIBUTES.put(attribute.id(), attribute);
    }

    public static Optional<Attribute> get(String id) {
        return Optional.ofNullable(ATTRIBUTES.get(id));
    }

    public static Collection<Attribute> getAll() {
        return ATTRIBUTES.values();
    }
}

File: src/main/java/net/reflact/common/attribute/AttributeRegistry.kt
package net.reflact.common.attribute

import java.util.Optional

object AttributeRegistry {
    private val ATTRIBUTES = HashMap<String, Attribute>()

    fun register(attribute: Attribute) {
        ATTRIBUTES[attribute.id] = attribute
    }

    fun get(id: String): Optional<Attribute> {
        return Optional.ofNullable(ATTRIBUTES[id])
    }

    fun getAll(): Collection<Attribute> {
        return ATTRIBUTES.values
    }
}
